primitives/hp-groth16/src/data_structures.rs

```diff
@@ -86,7 +86,7 @@ pub struct VerificationKey {
 impl G1 {
     /// Try to convert the G1 point to an affine representation.
     pub fn try_into_affine<R: AffineRepr>(self) -> Result<R, SerializationError> {
-        R::deserialize_uncompressed(self.0.as_ref())
+        R::deserialize_uncompressed(self.0.as_slice())
```

Comment:
Really we need it here. Moreover, why is better?

Reply:
I introduced this modification in https://github.com/HorizenLabs/zkVerify/pull/136/commits/fcf2da198320f9e3ac6a45bfdd1e8577d6f089d4 in order to solve a compilation error. Somehow, introducing the proof-of-sql-verifier (which also uses arkworks serialization) did confuse the compiler regarding type inference.

This fix was necessary up to commit https://github.com/HorizenLabs/zkVerify/pull/136/commits/3a1efba014899aba50adb76f77b1dafce78dbd8a. To reproduce, just checkout that commit and roll back the modification. Compilation fails.

Then, something in https://github.com/HorizenLabs/zkVerify/pull/136/commits/a594e16be0c79a328704d31a98518064a859bc9f (the commit in which we switch from the native version of the verifier to the wasm one) made the fix unnecessary for compilation.

So, to sum up, at the moment the code compiles fine even without this fix, but at some point the fix was necessary. I did not deep-dive into the issue, but maybe, given your greater experience with Rust, it's easy for you to spot the problem.

Reply:
My first guess here is some issue related to features about arkworks crates. The old groth16 add arkworks with some others features but this set of features is changed after the intruduction of this verifier?